### ISSUE TITLE
Search Features Won't Enable

### ISSUE THREAD

**Issue by BigPhobos** [mezz/JustEnoughItems#2555]
When I try to enable searching by color or creative tab nothing works. I can search by modid or tags but no matter how I set the config, it won't let me search for colors or creative tabs.
![image](https://user-images.githubusercontent.com/76231975/140576967-639b27ea-8785-4a2c-bde1-de8bfd5d4077.png)
![image](https://user-images.githubusercontent.com/76231975/140576975-0bd4a124-5b8d-41a4-88ca-3f0321486afe.png)

**Comment by mezz** [comment]
Thanks for the report!
I need some more information to solve this issue, please paste your full `fml-client-latest.log` or `latest.log` to https://gist.github.com and link it here.
And can you paste your config here?

**Comment by BigPhobos** [comment]
https://gist.github.com/LunarPhobos/db5163284ef59f0f276d584e1f71b5ba
```
[advanced]
	#Debug mode enabled
	DebugMode = false
	#Display search bar in the center
	CenterSearch = false
	#Set low-memory mode (makes search very slow, but uses less RAM)
	LowMemorySlowSearchEnabled = false
	#How items should be handed to you
	#Allowed Values: INVENTORY, MOUSE_PICKUP
	GiveMode = "MOUSE_PICKUP"
	#Max number of columns shown
	#Range: 4 ~ 100
	MaxColumns = 9
	#Max. recipe gui height
	#Range: > 175
	RecipeGuiHeight = 350

[colors]
	#Color values to search for
	SearchColors = ["White:EEEEEE", "LightBlue:7492cc", "Cyan:00EEEE", "Blue:2222dd", "LapisBlue:25418b", "Teal:008080", "Yellow:cacb58", "GoldenYellow:EED700", "Orange:d97634", "Pink:D1899D", "HotPink:FC0FC0", "Magenta:b24bbb", "Purple:813eb9", "JadedPurple:43324f", "EvilPurple:2e1649", "Lavender:B57EDC", "Indigo:480082", "Sand:dbd3a0", "Tan:bb9b63", "LightBrown:A0522D", "Brown:634b33", "DarkBrown:3a2d13", "LimeGreen:43b239", "SlimeGreen:83cb73", "Green:008000", "DarkGreen:224d22", "GrassGreen:548049", "Red:963430", "BrickRed:b0604b", "NetherBrick:2a1516", "Redstone:ce3e36", "Black:181515", "CharcoalGray:464646", "IronGray:646464", "Gray:808080", "Silver:C0C0C0"]

[sorting]
	#Sorting order for the ingredient list. Valid stages: [MOD_NAME, INGREDIENT_TYPE, ALPHABETICAL, CREATIVE_MENU, TAG, TOOL_TYPE, WEAPON_DAMAGE, ARMOR, MAX_DURABILITY]
	IngredientSortStages = ["MOD_NAME", "INGREDIENT_TYPE", "CREATIVE_MENU", "ALPHABETICAL", "WEAPON_DAMAGE", "TOOL_TYPE", "ARMOR", "TAG"]

[search]
	#Search mode for Mod Names (prefix: @)
	#Allowed Values: ENABLED, REQUIRE_PREFIX, DISABLED
	ModNameSearchMode = "REQUIRE_PREFIX"
	#Search mode for Tooltips (prefix: #)
	#Allowed Values: ENABLED, REQUIRE_PREFIX, DISABLED
	TooltipSearchMode = "ENABLED"
	#Search mode for Tag Names (prefix: $)
	#Allowed Values: ENABLED, REQUIRE_PREFIX, DISABLED
	TagSearchMode = "REQUIRE_PREFIX"
	#Search mode for Creative Tab Names (prefix: %)
	#Allowed Values: ENABLED, REQUIRE_PREFIX, DISABLED
	CreativeTabSearchMode = "ENABLED"
	#Search mode for Colors (prefix: ^)
	#Allowed Values: ENABLED, REQUIRE_PREFIX, DISABLED
	ColorSearchMode = "ENABLED"
	#Search mode for resources IDs (prefix: &)
	#Allowed Values: ENABLED, REQUIRE_PREFIX, DISABLED
	ResourceIdSearchMode = "DISABLED"
	#Search advanced tooltips (visible with F3+H)
	SearchAdvancedTooltips = false

[modname]
	#Formatting for mod name tooltip
	#Use these formatting keys:
	#black, dark_blue, dark_green, dark_aqua, dark_red, dark_purple, gold, gray, dark_gray, blue, green, aqua, red, light_purple, yellow, white
	#obfuscated, bold, strikethrough, underline, italic
	ModNameFormat = "blue italic"

```

**Comment by mezz** [comment]
Thank you, this should be fixed in the latest version of JEI for 1.16 and 1.17.
